django/db/models/expressions.py

```diff
@@ -519,7 +519,7 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
             c.source_expressions[pos] = arg.resolve_expression(query, allow_joins, reuse, summarize, for_save)
         return c
 
-    def as_sql(self, compiler, connection, function=None, template=None):
+    def as_sql(self, compiler, connection, **kwargs):
```

Comment:
Probably not a bad idea, but it does make the implementation slightly more complicated.